flink/src/main/java/org/apache/iceberg/flink/source/IcebergSourceEvents.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink.source;
+
+import java.util.Collection;
+import java.util.Collections;
+import javax.annotation.Nullable;
+import org.apache.flink.api.connector.source.SourceEvent;
+import org.apache.flink.api.connector.source.SourceReader;
+import org.apache.flink.api.connector.source.SplitEnumerator;
+
+/**
+ * These source events aren't included in Flink 1.11.2.
+ * Copying the definitions here.
+ *
+ * Once the latest flink-connector-base changes land in 1.11.3 or 1.12.0,
+ * we can remove at least NoMoreSplitsEvent and NoSplitAvailableEvent.
+ */
+public class IcebergSourceEvents {
+
+  /**
+   * A {@link SourceEvent} representing the request for a split, typically sent from the
+   * {@link SourceReader} to the {@link SplitEnumerator}.
+   *
+   * TODO: push change to Flink to carry the finished splitIds.
```

Comment:
Is there a JIRA for this?

Reply:
Yeah, I forgot to follow up on the Flink side. Created the jira and attached a PR to it.
https://issues.apache.org/jira/browse/FLINK-21364